### PR TITLE
fix model merge: train_args can be different in different ranks

### PR DESCRIPTION
**Pull request by donglixp**
# NOTE: train_args can be different in different ranks
            # for example, profiling related args can be different,
            # so we don't want to enforce them to be the same across ranks.

### CODE REVIEW

**Comment by 0xWJ** [comment]
Can you give me an example when model args and lr_scheduler args are different across ranks?

File: nnscaler/cli/trainer.py [hunk 1]
```diff
@@ -270,9 +270,9 @@ def _merge_checkpoint(cls, checkpoint_files: List[str],
             # for example, profiling related args can be different,
             # so we don't want to enforce them to be the same across ranks.
             if state_dicts[i]['train_args']['model'] != state_dicts[0]['train_args']['model']:
-                raise ValueError(f"model config in {checkpoint_files[i]} is different from {checkpoint_files[0]}")
+                print(f"model config in {checkpoint_files[i]} is different from {checkpoint_files[0]}")
             if state_dicts[i].get('lr_scheduler', None) != state_dicts[0].get('lr_scheduler', None):
-                raise ValueError(f"lr_scheduler state in {checkpoint_files[i]} is different from {checkpoint_files[0]}")
+                print(f"lr_scheduler state in {checkpoint_files[i]} is different from {checkpoint_files[0]}")
```
[the remark this answers is not in the document]
Comment (0xWJ) [reply nnscaler/cli/trainer.py]: Agree. Please replace print with logger.warning

**Comment by donglixp** [comment]
> Can you give me an example when model args and lr_scheduler args are different across ranks?

@0xWJ e.g., offiline merging

**Comment by 0xWJ** [comment]
> > Can you give me an example when model args and lr_scheduler args are different across ranks?
> 
> @0xWJ e.g., offiline merging

i mean in offline merging, why model config/lr scheduler data can be different? can you give me an example?
